projectdoc/architektur.tex

```diff
@@ -0,0 +1,139 @@
+
+\section{Architektur}
+\label{sec:Architektur}
+% Vorgehen nach C4 Modell
+In diesem Kapitel wird die Architektur der PlazaRoute Software und die Schnittstellen zu den Umsystemen besprochen. Als Anhaltspunkt wird das C4 Modell \cite{c4model} von Simon Brown verwendet. In einem ersten Schritt wird unsere Applikation in den Kontext des grösseren Systems gesetzt. Anschliessend teilen wir das System PlazaRoute in einzelne Container und den zentralen Container in einzelne Komponenten auf.
+
+\subsection{Systemkontext}
+\label{architektur:Systemkontext}
+
+\begin{figure}[ht]
+\centering
+% TODO: Grafik zuschneiden
+\includegraphics[width=1\linewidth]{projectdoc/img/system-context_diagram.png}
+\caption[System Kontext Diagramm]{System PlazaRoute im Kontext mit Umsystemen; Grafik erstellt mit \emph{Structurizr Express}\cite{structurizr}}
+\label{fig:system_context_diagram}
+\end{figure}
+
+Abbildung \ref{fig:system_context_diagram} zeigt das System PlazaRoute mit den Umsystemen auf.
+
+Der User bedient die QGIS Desktop Applikation mit dem von uns entwickelten Plugin. Dieses leitet die die Eingabe der Start- und Zielkoordinaten an das System PlazaRoute weiter. Als Antwort sendet PlazaRoute eine Routenbeschreibung an das QGIS-Plugin zurück, das es im QGIS dargestellt.
+
+\subsection{Container}
+\label{architektur:Container}
+
+\begin{figure}[ht]
+    \centering
+    % TODO: Grafik zuschneiden
+    \includegraphics[width=1\linewidth]{projectdoc/img/container_diagram.png}
+    \caption[Container Diagramm]{Container Diagramm von PlazaRoute; Grafik erstellt mit \emph{Structurizr Express}\cite{structurizr}}
+    \label{fig:container_diagram}
+    \end{figure}    
+
+In Abbildung \ref{fig:container_diagram} zoomen wir in das System PlazaRoute hinein und teilen es in drei Container auf, die logisch voneinander getrennt sind. So könnten die Container auch verteilt deployed werden.
+
+In den nachfolgenden Abschnitten werden die drei Container näher beleuchtet.
```

Comment:
@Excape es ist die Rede von drei Container. Aber das QGIS-Plugin gehört nicht zu diesen drei Container oder? Ist es dann noch am richtigen Ort? Oder soll es woanders hin?

Reply:
Vlt könnten wir auch ein Kapitel Umsysteme reinnehmen und dies mit search.ch, osm und overpass beschreiben?

Reply:
In der Grafik wird Routing Engine geschrieben, im Text ist von Routing-Engine die Rede

Reply:
Hmm, habe nun das QGIS-Plugin aus Container heraus genommen. Aus Sicht von C4 ist das QGIS-Plugin ein System mit einem Container. Wenn wir die Umsysteme in einem eigenen Kapitel betrachten, nehmen wir sie von den inneren Komponenten weg, was m.M.n etwas unübersichtlich wäre